Tests/Shared.Specs/ExecutionTimeAssertionsSpecs.cs

```diff
@@ -94,6 +94,33 @@ public void When_the_execution_time_of_an_action_is_less_or_equal_to_a_limit_it_
             //-----------------------------------------------------------------------------------------------------------
             act.Should().NotThrow();
         }
+
+        [Fact]
+        public void When_action_runs_indefinitely_it_should_be_stopped_and_throw_if_there_is_less_or_equal_condition()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            Action someAction = () =>
+            {
+                // lets cause a deadlock
+                var semaphore = new Semaphore(0, 1); // my weapon of choice is a semaphore
+                semaphore.WaitOne(); // oops
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => someAction.ExecutionTime().Should().BeLessThan(100.Milliseconds());
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act
+                .Should().Throw<XunitException>()
+                .And.Message.Should().StartWith(
```

Comment:
Suggestion: use `WithMessage`.
E.g. `WithMessage("*action should be less than 0.100s, but it required*")`.

Reply:
nice, did not know about this one